### PR TITLE
CPM-782: Create controller and rest route to GET an identifier generator

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/back/src/Infrastructure/Controller/GetIdentifierGeneratorController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GetIdentifierGeneratorController
```
Comment (antoinearrive) [inline components/identifier-generator/back/src/Infrastructure/Controller/GetIdentifierGeneratorController.php:17]:
```suggestion
final class GetIdentifierGeneratorController
```

**Review by antoinearrive — APPROVED** [review]

File: components/identifier-generator/back/tests/EndToEnd/Infrastructure/Controller/GetIdentifierGeneratorControllerEndToEnd.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Pim\Automation\IdentifierGenerator\EndToEnd\ControllerEndToEndTestCase;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GetIdentifierGeneratorControllerEndToEnd extends ControllerEndToEndTestCase
```
Comment (antoinearrive) [inline components/identifier-generator/back/tests/EndToEnd/Infrastructure/Controller/GetIdentifierGeneratorControllerEndToEnd.php:16]:
```suggestion
final class GetIdentifierGeneratorControllerEndToEnd extends ControllerEndToEndTestCase
```

**Review by antoinearrive — APPROVED** [review]

File: components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/routing.yml [hunk 3]
```diff
@@ -10,3 +10,8 @@ akeneo_identifier_generator_rest_create:
     path: '/rest/identifier-generator'
     controller: Akeneo\Pim\Automation\IdentifierGenerator\Infrastructure\Controller\CreateIdentifierGeneratorController
     methods: [POST]
+
+akeneo_identifier_generator_rest_get:
+    path: '/rest/identifier-generator/{code}'
+    controller: Akeneo\Pim\Automation\IdentifierGenerator\Infrastructure\Controller\GetIdentifierGeneratorController
+    methods: [GET]
```
Comment (Beninho) [inline components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/routing.yml:17]: What do you think about a regex to validate code format ?

**Review by Beninho — APPROVED** [review]

File: components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/routing.yml (hunk 3 above)
Comment (Balamung) [reply components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/routing.yml:17]: Yup ! Something like /^[a-zA-Z0-9_]+$/ should be good, or do we have specifics in this one @pierallard ?

Comment (pierallard) [reply components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/routing.yml:17]: I think this is not needed, as there are no other routes /rest/identifier-generator/something accessible via GET. You will have a better 404 if you don't put regexp.

**Pull request merged**